### ISSUE TITLE
Self-contradictory Type Mismatch error

### ISSUE THREAD

**Issue by speezepearson** [elm/compiler#2124]
**Quick Summary:** When trying to treat a tuple-containing-something-non-`comparable` as a `comparable` (as far as I can tell), the error message says, `This argument is a tuple of type: <something> But yourFunc needs the first argument to be <the exact same thing>`.

## SSCCE

```elm
---- Elm 0.19.0 ----------------------------------------------------------------
Read <https://elm-lang.org/0.19.0/repl> to learn more: exit, help, imports, etc.
--------------------------------------------------------------------------------
> List.sort [(1, {})]
-- TYPE MISMATCH ----------------------------------------------------------- elm

The 1st argument to `sort` is not what I expect:

4|   List.sort [(1, {})]
               ^^^^^^^^^
This argument is a list of type:

    List ( number, {} )

But `sort` needs the 1st argument to be:

    List ( number, {} )

```

- **Elm:** 0.19.0
- **Browser:** n/a
- **Operating System:** Ubuntu 18.04

## Additional Details

- This isn't just a REPL problem, it occurs in  `elm reactor` as well.
- [But not, interestingly, in Ellie.](https://ellie-app.com/9j9h5nBrC8sa1)
- `type MyType = Foo ; List.sort [(1, Foo)]` has this same issue.
- `f : comparable -> () ; f _ = () ; f (1, {})` has this same issue.
- `x : comparable ; x = (1, {})` does _not_ have this issue.

**Comment by zwilias** [comment]
This was fixed in 0.19.1, which is why you're not seeing it in Ellie:

```
---- Elm 0.19.1 ----------------------------------------------------------------
Say :help for help and :exit to exit! More at <https://elm-lang.org/0.19.1/repl>
--------------------------------------------------------------------------------
> List.sort [(1, {})]
-- TYPE MISMATCH ---------------------------------------------------------- REPL

The 1st argument to `sort` is not what I expect:

3|   List.sort [(1, {})]
               ^^^^^^^^^
This argument is a list of type:

    List ( number, {} )

But `sort` needs the 1st argument to be:

    List comparable

Hint: I only know how to compare ints, floats, chars, strings, lists of
comparable values, and tuples of comparable values.
```

**Comment by speezepearson** [comment]
Aha! Lovely. I just upgraded and this is indeed fixed. Thanks, @zwilias!
